### ISSUE TITLE
MT32 emulation not detected in macOS build of 0.77.1 using Homebrew

### ISSUE THREAD

**Issue by almeath** [dosbox-staging/dosbox-staging#1270]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### What Operating System are you using?

Mac OS

### Whats your question and how can we help?

I built the latest version of DOSBox Staging using Homebrew on macOS Mojave 10.14.6. 

While Fluidynth and sound fonts work fine, when I set the midi device to mt32 I see the following error in the Terminal output:

"mt32" is not a valid value for variable: midi device

I looked in the wiki and also tried adding parameters for "model" and "romdir" and then additional errors appear in the Terminal output:

CONFIG: Unknown option model
CONFIG: Unknown option romdir

It seems the Homebrew build command is not building 0.77.1 with the MT32emu component included. However, I cannot test the fully compiled downloadable version because I am not using macOS 10.15 or higher.

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by kcgen** [comment]
Thanks for this heads up @almeath. That's too bad Brew isn't giving users the full functionality; mt32 emulation is a big loss!

I see the problem is on line 27 of Brew's ruby manifest: <https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/dosbox-staging.rb#L27>

Can you open a bug with the brew team?

Instead of disabling mt32emu, I suggest using `--force-fallback-for=mt32emu` in it's place, which should enable it.

**Comment by almeath** [comment]
Thanks for confirming where the bug is in Homebrew. I will look into filing a report.

**Comment by kcgen** [comment]
@almeath - if you have a link to the Brew ticket handy, please link it here, which wil l notify us when brew solves it.

**Comment by carlocab** [comment]
What does `--force-fallback-for=mt32emu` do? We pass `-Duse_mt32emu=false` because the build fails without it. See Homebrew/homebrew-core#80510 for context.

**Comment by kcgen** [comment]
> What does `--force-fallback-for=mt32emu` do? 

It will allow the DOSBox Staging Meson build to use internally provided mt32emu wrap to satisfy this dependency. 

Please try adding it, and you will see the build should work without issues. 

but it seems Homebrew needs to ship `mt32emu` first

This is not the case; please try the above. Thanks @carlocab

**Comment by kcgen** [comment]
I originally tagged this with `external` because it's outside our control.

For context, we tag issues with **external** for 3rd party dependencies that affect the health of Staging itself (SDL2/Loguru/Audio codecs/...).

In this case, the `external` tag was a bit generous because this is outside our scope and not a dependency. Likewise, repository packages are managed by those teams.

@almeath , I hope this is solved, or will be soon! 

I am closing this, so if needed please keep following up with the Homebrew team.

**Comment by carlocab** [comment]
> > but it seems Homebrew needs to ship `mt32emu` first
> 
> This is not the case; please try the above. Thanks @carlocab

Actually, this is the case. We pass `--wrapmode=nofallback` intentionally: https://github.com/Homebrew/brew/commit/6af2b6db3113a3bc6e500f421de72d7fedc37b4e

@almeath, if you're still interested in this, please submit a pull request to add `mt32emu` to Homebrew, and we can add support for it in the `dosbox-staging` formula.

**Comment by kcgen** [comment]
@carlocab,

Can you please try the following line in place of your existing `std_meson_args`?

I believe it will produce a fast and static package, without any need for a separate `mt32emu` package (or FluidSynth, for that matter).

``` python
def std_meson_args
    ["--prefix=#{prefix}", "--libdir=#{lib}", "--buildtype=release", "--wrap-mode=forcefallback", "-Ddefault_library=static", "-Db_asneeded=true", "-Db_lto=true", "-Dtry_static_libs=fluidsynth,mt32emu,opusfile,png,sdl2,sdl2_net", "-Dfluidsynth:enable-floats=true", "-Dfluidsynth:try-static-deps=true"]
```

**Comment by carlocab** [comment]
That will probably work, but that goes against our packaging guidelines. We link dependencies dynamically on purpose.

**Comment by carlocab** [comment]
@almeath, this should be fixed by Homebrew/homebrew-core#90175.

**Comment by almeath** [comment]
Thanks, but it looks like a bit of a Catch 22 situation now. I have had problems getting Homebrew to update on Mojave, and it is stated that 10.14 is no longer supported. The build command for DOSBox Staging now fails, and so I do not end up with a binary to test. However I do see in the output that the munt component is being called. Since I only went down this path to try to get a fully featured DOSBox-Staging working with Mojave, I may have to call it quits and upgrade now. If Homebrew is no longer supporting the older OS then that is a much bigger issue for me than Staging not being available with MT-32 support. Thanks again for your efforts in tracking down this issue and getting it resolved though.

**Comment by kcgen** [comment]
> The build command for DOSBox Staging now fails, and so I do not end up with a binary to test.

That's not good. Do you think it's due to homebrew issues? Can you attach the log of a build attempt?

**Comment by carlocab** [comment]
Try removing `-Db_lto=true` from the formula and see if that helps. Mojave has an old-ish toolchain so it might struggle with LTO.

**Comment by almeath** [comment]
Looks like I was having some problem with the Xcode command line tools. I removed it entirely and re-installed version 11.3.1 (the latest that is compatible with macOS 10.14), after which the DOSBox Staging build succeeded. I did not have to remove the above command from the formula. All tested and working perfectly now with MT-32 support. Thanks again for your help @carlocab and @kcgen , this issue can be left safely closed and considered resolved.
